include/magma/vk/buffers/BufferManager.hpp

```diff
@@ -0,0 +1,40 @@
+#include "magma/vk/buffers/BufferManager.h"
+template<class T>
+Buffer& BufferManager::createBufferWithData(const std::string &name, const std::vector<T> &data, 
+        VkBufferUsageFlags usage, VkMemoryPropertyFlags properties) {
+    if (bufferExists(name)) {
+        throw std::invalid_argument("BufferManager::createBufferWithData buffer exist");
+    }
+
+    VkDeviceSize dataSize = sizeof(data[0]) * data.size();
+    return createBufferWithData(name, data.data(), dataSize, usage, properties);
+}
+
+template<class T>
+Buffer& BufferManager::createVertexBuffer(const std::string &name, const std::vector<T> &data) {
+    if (bufferExists(name)) {
```

Comment:
oh, this check is executed 4 times for this function!!!

Reply:
Done